evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -219,38 +334,57 @@ def test_stl_decomposer_inverse_transform(
 @pytest.mark.parametrize("fit_before_decompose", [True, False])
 def test_stl_decomposer_get_trend_dataframe(
     generate_seasonal_data,
+    generate_multiseries_seasonal_data,
     transformer_fit_on_data,
     fit_before_decompose,
     variateness,
 ):
     period = 7
-    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
-        period=period,
-        freq_str="D",
-        set_time_index=True,
-    )
+
+    if variateness == "univariate":
+        X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+            period=period,
+            freq_str="D",
+            set_time_index=True,
+        )
+        subset_y = y[: 5 * period]
+    elif variateness == "multivariate":
+        X, y = generate_multiseries_seasonal_data(real_or_synthetic="synthetic")(
+            period=period,
+            freq_str="D",
+            set_time_index=True,
+        )
```

Comment:
Alright, this code block has shown up so many times, I think it needs its own function in `conftest` - something where we just pass in whether the problem is univariate or multivariate, and it returns the correct data for us.